src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -615,15 +615,14 @@ public static string TryGetAndroidJarPath (TaskLoggingHelper log, string platfor
 			return Path.Combine (platformPath, "android.jar");
 		}
 
-		public static Dictionary<string, string> LoadResourceCaseMap (string resourceCaseMap)
-		{
-			var result = new Dictionary<string, string> ();
-			if (resourceCaseMap != null) {
-				foreach (var arr in resourceCaseMap.Split (';').Select (l => l.Split ('|')).Where (a => a.Length == 2))
-					result [arr [1]] = arr [0]; // lowercase -> original
-			}
-			return result;
-		}
+		static readonly string ResourceCaseMapKey = $"{nameof (MonoAndroidHelper)}_ResourceCaseMap";
+
+		public static void SaveResourceCaseMap (IBuildEngine4 engine, Dictionary<string, string> map) =>
+			engine.RegisterTaskObject (ResourceCaseMapKey, map, RegisteredTaskObjectLifetime.Build, allowEarlyCollection: false);
+
+		public static Dictionary<string, string> LoadResourceCaseMap (IBuildEngine4 engine) =>
+			engine.GetRegisteredTaskObject (ResourceCaseMapKey, RegisteredTaskObjectLifetime.Build)
+				as Dictionary<string, string> ?? new Dictionary<string, string> (0);
```

Comment:
does the empty value need to include a StringComparer like we do when we created it at https://github.com/xamarin/xamarin-android/pull/5188/files#diff-a154f765074e2f487da8dc8a1fd28ac3R76

Reply:
If the `??` evaluates, then something really went wrong. I thought an empty dictionary would be better than a `null`.

It could `LogError` instead, but it means we have something broken not-actionable for users. Should it be an error instead?

Reply:
erring out with a non actionable error is probably not the best. Let's leave this as it is.